TicTacToeApp/ComputerPlayer.swift

```diff
@@ -16,10 +26,48 @@ public class ComputerPlayer {
       }
     }
 
-    return getRandomElement(availableSpaces)
+    return availableSpaces
   }
 
   private func getRandomElement(array: [Int]) -> Int {
     return array[Int(arc4random_uniform(UInt32(array.count)))]
   }
+
+  private func makeBlockingRequest(request: NSURLRequest, onSuccess: (String) -> Void) {
+    let semaphore = dispatch_semaphore_create(0)
```

Comment:
Without the semaphore, ComputerPlayer#makeMove() returns before the "move" variable can be initialized in the callback passed to #makeBlockingRequest(). After I change the design by removing the return value from #makeMove() and fully delegating the computer move to an asynchronous callback, it shouldn't be necessary.